app/models/user.rb

```diff
@@ -107,6 +107,11 @@ def inactive_message
     "Sorry, this account has been disabled."
   end
 
+  # Returns all users who match the query.
+  def self.search_from_query(members, query)
+    User.enabled.where.not(id: members).where(arel_table[:username].matches(query)).pluck(:username)
```

Comment:
I agree, but I still see scopes as "handy one-liners for models" to put it in a way. I guess it's a matter of different tastes :) I don't have any strong opinions between 2 vs 3, but if I had to vote for any of these, I'd vote the second option.